rspace/src/main/scala/coop/rchain/rspace/internal.scala

```diff
@@ -23,7 +23,12 @@ object internal {
       Datum(a, persist, Produce.create(channel, a, persist, sequenceNumber))
   }
 
-  final case class DataCandidate[C, A](channel: C, datum: Datum[A], datumIndex: Int)
+  final case class DataCandidate[C, A](
+      channel: C,
+      datum: Datum[A],
+      removedDatum: A,
```

Comment:
A hunch tells me that this field is redundant. I cannot check this because tests are currently failing. Will get back to reviewing once this is sorted out

Reply:
You can believe me that the distinction between `removed` and `matched` is part of the non-symmetric costs problem.

Reply:
@lukasz-golebiewski the tests are now passing, please have another stab at it

Reply:
You can verify the need for removed datum by using the test I've added in CostAccountingSpec in https://github.com/rchain/rchain/pull/2682/commits/763bfd35cabd1293a11467879a1ae22fb1649435

Reply:
Did it, thanks!